joomla/joomla-cms
# [4.0] Option to use a menu item for privacy policy.

This PR is based on a feature request brought up by user fire67 on the German forum: https://forum.joomla.de/thread/11738-plugin-inhalt-datenschutzerkl%C3%A4rung/

Apparently, some page builders like SP PageBuilder don't register their "articles" as real Joomla! articles. If you create your privacy policy using such a page builder (might apply to some CCK extensions, too), you can't choose it in the privacy consent plugin, because there, only articles are available.

### Summary of Changes
This PR adds a menu item parameter to plg_system_privacyconsent and plg_content_confirmconsent that can be used instead of the article parameter, and a parameter to choose between the two.

### Testing Instructions
At least one test on a multilingual site would be very welcome. I replicated the mechanism that chooses the associated article in the respective language, for menu items. It works on my small local setup. But as I'm no expert in multilingual sites, I'd be happy if someone with more expertise could check if it works as expected.

1. Enable the Plugin "System - Privacy Consent".
2. Select an article for your privacy policy.
3. In frontend, navigate to the registration form and make sure that the modal box for the privacy policy works as expected (in all languages).
4. Apply this PR.
5. Make sure everything still works like before.
6. In the plugin config, switch "Privacy Type" to "Menu Item" and select a menu item. To make sure that there is a difference, maybe select a category blog or something else, but not the menu item for the privacy article you used before.
7. Repeat step 3.
8. Switch the type back to "Article" and make sure it works as before.
9. Repeat everything from 1 to 8 with the plugin "Content - Confirm Consent" and the contact form.

You might want to check if the privacy links work for all settings with SEO URLs enabled and disabled.

### Documentation Changes Required
Should be documented at https://docs.joomla.org/J3.x:Privacy (or the J4 version of that?).

### Additional Notes
I noticed that the full `$article` object including link, alias, id, catid and language, is passed to the layout, but only the link is used there (at least in core). Should I do the same for the menu item, so overrides could access further information or leave it as it is?
